src/Octoshift/GithubClient.cs

```diff
@@ -73,6 +73,27 @@ public virtual async IAsyncEnumerable<JToken> GetAllAsync(string url, Dictionary
         public virtual async Task<string> PostAsync(string url, object body, Dictionary<string, string> customHeaders = null) =>
             (await SendAsync(HttpMethod.Post, url, body, customHeaders: customHeaders)).Content;
 
+        public virtual async Task<JObject> PostGraphQLAsync(
+            string url,
+            object body,
+            Dictionary<string, string> customHeaders = null)
+        {
+            var jBody = JObject.FromObject(body);
+            jBody["variables"] ??= new JObject();
+
+            var (content, _) = await SendAsync(HttpMethod.Post, url, jBody, customHeaders: customHeaders);
+            var response = JObject.Parse(content);
+
+            if (response.TryGetValue("errors", out var jErrors) && jErrors is JArray { Count: > 0 } errors)
+            {
+                var error = (JObject)errors[0];
+                var errorMessage = error.TryGetValue("message", out var jMessage) ? (string)jMessage : null;
+                throw new OctoshiftCliException($"{errorMessage ?? "UNKNOWN"}");
+            }
+
+            return response;
+        }
```

Comment:
This method basically needs to call the `PostAsync` and then additionally ensures a success GQL response. So no need to re-set the `variables` or do anything extra (it does so in the paginated version for a reason), it should be as simple as passing in an object body and then re-passing it to `PostAsync`. I would also recommend keeping the error checking logic in its separate method since it also needs to be called in the `PostGraphQLWithPaginationAsync()` method. With all these your method should look like this (Note that the method return a `JToken` which is more generic that `JObject`):
```suggestion
        public virtual async Task<JToken> PostGraphQLAsync(
            string url,
            object body,
            Dictionary<string, string> customHeaders = null)
        {
            var response = await PostAsync(url, body, customHeaders);
            var data = JObject.Parse(response);

            EnsureSuccessGraphQLResponse(data);
            
            return data;
        }
```

---
### The following can also be tackled in a separate PR

Also the `PostGraphQLWithPaginationAsync()` needs to be refactored with the above changes, just two lines of changes (use `PostAsync` instead of `SendAsync` and call `EnsureSuccessGraphQLResponse()`):
```c#
public virtual async IAsyncEnumerable<JToken> PostGraphQLWithPaginationAsync(...)
{
    ...

    var response = await PostAsync(url, body, customHeaders);
    var jContent = JObject.Parse(content);
    EnsureSuccessGraphQLResponse(jContent);

    ...
}
```
of course a test also needs to be added for `PostGraphQLWithPaginationAsync()` to make sure that it throws in case of a GQL error.

Reply:
Thanks for the feedback! Totally makes sense - and interesting to know that JToken is more generic, I didn't know that 😊

I'll update this tomorrow - and can look at the paginated method in a separate PR.